### PR TITLE
Add Bitcoin payment method

### PR DESCRIPTION
**Pull request by geco**

### CODE REVIEW

File: config/settings.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "cartTitle": "CLOTH",
+    "cartTitle": "GECO DEV CLOTH",
```
Comment (mrvautin) [inline config/settings.json]: Pls revert this.

File: config/settings.json [hunk 2]
```diff
@@ -17,13 +17,17 @@
     "googleAnalytics": "",
     "injectJs": "",
     "customCss": "",
-    "currencySymbol": "£",
-    "paymentGateway": "stripe",
+    "currencySymbol": "$",
```
Comment (mrvautin) [inline config/settings.json]: Please change back to default

File: config/settings.json [hunk 3]
```diff
@@ -17,13 +17,17 @@
...
+    "currencyISO": "USD",
+    "paymentGateway": "blockonomics",
```
Comment (mrvautin) [inline config/settings.json]: Please change back to default

File: config/settings.json [hunk 4]
```diff
@@ -32,6 +36,14 @@
...
             "discount": "discount-voucher"
+        },
+        "loaded": {
+            "shipping": {},
+            "discount": {}
         }
-    }
-}
+    },
+    "secretCookie": "378f764ea5992d9e910b1868b3a049874eac5829",
```
Comment (mrvautin) [inline config/settings.json]: Please remove secrets

File: package.json [hunk 5]
```diff
@@ -72,6 +72,7 @@
     "strip-bom": "^3.0.0",
     "stripe": "^7.12.0",
     "uglifycss": "0.0.27",
+    "unirest": "^0.6.0",
```
Comment (mrvautin) [inline package.json]: Please use `Axios` as the "http" type module

File: routes/payments/blockonomics.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+                indexOrders(req.app)
+                .then(() => {
+                  
+                    // set the order ID in the session, to link to it from blockonomics payment page
+                    blockonomicsParams.pendingOrderId = newId;
+                    req.session.blockonomicsParams = blockonomicsParams;
+                    res.redirect('/blockonomics_payment');
+                    
+                });
+
```
Comment (mrvautin) [inline routes/payments/blockonomics.js]: Pls remove whitespace

File: .gitignore [hunk 7]
```diff
@@ -5,4 +5,5 @@ public/uploads
 .vscode
 **.DS_Store
 env.yaml
-ecosystem.config.js
\ No newline at end of file
+ecosystem.config.js
+nohup.out
```
Comment (mrvautin) [inline .gitignore]: Whats this?

File: config/baseSchema.json [hunk 8]
```diff
@@ -69,15 +69,16 @@
         },
         "currencySymbol": {
             "type": "string",
-            "default": "£"
+            "default": "$"
```
Comment (mrvautin) [inline config/baseSchema.json]: Please revert

File: routes/payments/blockonomics.js [hunk 9]
```diff
@@ -0,0 +1,149 @@
+const express = require('express');
+const common = require('../../lib/common');
+const { indexOrders } = require('../../lib/indexing');
+const router = express.Router();
+const unirest = require('unirest');
+
```
Comment (mrvautin) [inline routes/payments/blockonomics.js]: Please remove whitespace and run linting checks

File: routes/payments/blockonomics.js [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+        
+            const orderDoc = {
+                orderPaymentId: blockonomicsParams.address,
+                orderPaymentGateway: 'Blockonomics',
+                orderExpectedBtc: blockonomicsParams.expectedBtc,
+                orderTotal: req.session.totalCartAmount,
+                orderShipping: req.session.totalCartShipping,
+                orderItemCount: req.session.totalCartItems,
+                orderProductCount: req.session.totalCartProducts,
+                orderEmail: req.session.customerEmail,
```
Comment (mrvautin) [inline routes/payments/blockonomics.js:97]: Please add the new field using the following line below line 105:

`
orderCompany: req.session.customerCompany,
`

**Review by mrvautin — CHANGES_REQUESTED** [review]
Look ok. Some small changes requested. There are many linting issues. Please run `npm run lint` and ensure there are no issues.

File: routes/payments/blockonomics.js (hunk 10 above)
Comment (geco) [reply routes/payments/blockonomics.js:97]: What's order company? Which new field?

File: config/settings.json (hunk 1 above)
Comment (geco) [reply config/settings.json]: Done

File: .gitignore (hunk 7 above)
Comment (geco) [reply .gitignore]: restored the original one

File: config/baseSchema.json (hunk 8 above)
Comment (geco) [reply config/baseSchema.json]: Restored original one

**Pull request merged**